tests/test_consumer.py

```diff
@@ -0,0 +1,158 @@
+        topic=joint_fixture.config.searchable_resource_events_topic,
+        key=f"dataset_embedded_{resource.id_}",
+    )
+
+    # consume the event
+    consumer = await joint_fixture.container.event_subscriber()
+    await consumer.run(forever=False)
+
+    # verify that the resource was added
+    updated_resources = await query_handler.handle_query(
+        class_name="DatasetEmbedded", query="", filters=[]
+    )
+    if is_insert:
+        assert updated_resources.count - results_all.count == 1
+    else:
+        assert updated_resources.count == results_all.count
+
+    assert resource in updated_resources.hits
+    assert resource not in results_all.hits
+
+
+@pytest.mark.asyncio
+async def test_resource_delete(joint_fixture: JointFixture):
+    """Test resource deletion via event consumption"""
+    query_handler = await joint_fixture.container.query_handler()
+
+    # get all the documents in the collection
+    targeted_initial_results = await query_handler.handle_query(
+        class_name="DatasetEmbedded",
+        query='"1HotelAlpha-id"',
+        filters=[],
+    )
+    assert targeted_initial_results.count == 1
+    assert targeted_initial_results.hits[0].id_ == "1HotelAlpha-id"
+
+    resource_info = event_schemas.SearchableResourceInfo(
+        accession="1HotelAlpha-id", class_name="DatasetEmbedded"
+    )
+
+    await joint_fixture.kafka.publish_event(
+        payload=resource_info.dict(),
+        type_=joint_fixture.config.resource_deletion_event_type,
+        topic=joint_fixture.config.searchable_resource_events_topic,
+        key=f"dataset_embedded_{resource_info.accession}",
+    )
+
+    # consume the event
+    consumer = await joint_fixture.container.event_subscriber()
+    await consumer.run(forever=False)
+
+    # get all the documents in the collection
+    results_post_delete = await query_handler.handle_query(
+        class_name="DatasetEmbedded", query='"1HotelAlpha-id"', filters=[]
+    )
+
+    assert results_post_delete.count == 0
+
+
+@pytest.mark.asyncio
+async def test_delete_nonexistent_resource(joint_fixture: JointFixture):
+    """Test for correct error handling when trying to delete a non-existent resource"""
+
+    query_handler = await joint_fixture.container.query_handler()
+
+    # get all the documents in the collection
+    all_results = await query_handler.handle_query(
+        class_name="DatasetEmbedded",
+        query="",
+        filters=[],
+    )
+
+    assert all_results.count > 0
+
+    resource_info = event_schemas.SearchableResourceInfo(
+        accession="notthere", class_name="DatasetEmbedded"
+    )
+
+    await joint_fixture.kafka.publish_event(
+        payload=resource_info.dict(),
+        type_=joint_fixture.config.resource_deletion_event_type,
+        topic=joint_fixture.config.searchable_resource_events_topic,
+        key=f"dataset_embedded_{resource_info.accession}",
+    )
+
+    # consume the event
+    with pytest.raises(DeletionFailedError):
```

Comment:
Actually, we don't want the consumer to throw an error (and thereby exiting the process). It should just log an error or warning and continue. The test could just check that something was logged.

Similar for the upsertion: If there was an error during the upsert, this should be just logged as an error, but not raise an exception.